update/process/baseProcess.go

```diff
@@ -155,60 +224,72 @@ func (b *baseProcessor) saveAllBlockDataToStorageForSelfShard(
+			log.Warn("saveMiniBlocks.Marshal -> MiniBlock",
 				"error", errNotCritical.Error())
 			continue
 		}
 
 		errNotCritical = b.storage.Put(dataRetriever.MiniBlockUnit, miniBlockHeadersHashes[i], marshalizedMiniBlock)
 		if errNotCritical != nil {
-			log.Warn("SaveAllBlockDataToStorageForSelfShard.Put -> MiniBlockUnit",
+			log.Warn("saveMiniBlocks.Put -> MiniBlockUnit",
 				"error", errNotCritical.Error())
 		}
 	}
+}
 
+func (b *baseProcessor) saveReceipts(headerHandler data.HeaderHandler) {
 	marshalizedReceipts, errNotCritical := b.txCoordinator.CreateMarshalizedReceipts()
 	if errNotCritical != nil {
-		log.Warn("SaveAllBlockDataToStorageForSelfShard.CreateMarshalizedReceipts",
+		log.Warn("saveReceipts.CreateMarshalizedReceipts",
 			"error", errNotCritical.Error())
-	} else {
-		if len(marshalizedReceipts) > 0 {
-			errNotCritical = b.storage.Put(dataRetriever.ReceiptsUnit, headerHandler.GetReceiptsHash(), marshalizedReceipts)
-			if errNotCritical != nil {
-				log.Warn("SaveAllBlockDataToStorageForSelfShard.Put -> ReceiptsUnit",
-					"error", errNotCritical.Error())
-			}
+		return
+	}
+
+	if len(marshalizedReceipts) > 0 {
+		errNotCritical = b.storage.Put(dataRetriever.ReceiptsUnit, headerHandler.GetReceiptsHash(), marshalizedReceipts)
+		if errNotCritical != nil {
+			log.Warn("saveReceipts.Put -> ReceiptsUnit",
+				"error", errNotCritical.Error())
 		}
 	}
+}
 
+func (b *baseProcessor) saveTransactions(body *block.Body) {
 	mapTxs := b.importHandler.GetTransactions()
 	for _, miniBlock := range body.MiniBlocks {
 		for _, txHash := range miniBlock.TxHashes {
 			tx, ok := mapTxs[string(txHash)]
 			if !ok {
-				log.Warn("SaveAllBlockDataToStorageForSelfShard",
+				log.Warn("saveTransactions",
 					"error", update.ErrTransactionNotFoundInImportedMap)
 				continue
 			}
 
-			unitType := getUnitTypeFromMiniBlockType(miniBlock.Type)
+			unitType, errNotCritical := getUnitTypeFromMiniBlockType(miniBlock.Type)
+			if errNotCritical != nil {
+				log.Warn("saveTransactions.getUnitTypeFromMiniBlockType",
+					"error", errNotCritical.Error())
+				continue
+			}
+
 			marshaledData, errNotCritical := b.marshalizer.Marshal(tx)
 			if errNotCritical != nil {
-				log.Warn("SaveAllBlockDataToStorageForSelfShard.Marshal -> Transaction",
+				log.Warn("saveTransactions.Marshal -> Transaction",
 					"error", errNotCritical.Error())
 				continue
 			}
 
 			errNotCritical = b.storage.Put(unitType, txHash, marshaledData)
 			if errNotCritical != nil {
-				log.Warn("SaveAllBlockDataToStorageForSelfShard.Put -> Transaction",
+				log.Warn("saveTransactions.Put -> Transaction",
 					"error", errNotCritical.Error())
 			}
 		}
 	}
 }
 
-func getUnitTypeFromMiniBlockType(mbType block.Type) dataRetriever.UnitType {
+func getUnitTypeFromMiniBlockType(mbType block.Type) (dataRetriever.UnitType, error) {
+	var err error
 	unitType := dataRetriever.TransactionUnit
 	switch mbType {
 	case block.TxBlock:
```

Comment:
`case block.TxBlock, block.InvalidBlock:` ?

Reply:
Actually this situation should not exist, as miniblock with type InvalidBlock is created only inside the self shard, and has sender = destination = self shard. In after hard fork process only cross mbs with dest in self shard are processed, as only cross shard mbs are found in unfinished meta blocks. Rather this should be signal as an error. So I think that with the current implementation the method will do the right thing if this type will be found. It will return ErrInvalidMiniBlockType.

Reply:
right